src/Components/Common/AzureComponent.cs

```diff
@@ -53,7 +53,10 @@ internal void AddClient(
         var configSection = builder.Configuration.GetSection(configurationSectionName);
 
         var settings = new TSettings();
+        // Bind both top-level and named configuration sections to the settings object
+        // to allow connection-specific settings.
         BindSettingsToConfiguration(settings, configSection);
+        BindSettingsToConfiguration(settings, configSection.GetSection(connectionName));
```

Comment:
This method is called for both keyed and non-keyed scenarios. When it is a `keyed` scenario, the `configSection` already has the `connectionName` in the config section. So in those cases this will bind to:

```
Aspire:Azure:Storage:Blobs:myBlob:myBlob
```